frontend/src/services/actions.ts

```diff
@@ -75,115 +79,19 @@ function getRiskText(risk: ActionSecurityRisk) {
 }
 
 export function handleActionMessage(message: ActionMessage) {
-  if ("args" in message && "security_risk" in message.args) {
-    store.dispatch(appendSecurityAnalyzerInput(message));
+  if (message.args?.hidden) {
+    return
   }
 
-  if (
-    (message.action === ActionType.RUN ||
-      message.action === ActionType.RUN_IPYTHON) &&
-    message.args.confirmation_state === "awaiting_confirmation"
-  ) {
-    if (message.args.thought) {
-      store.dispatch(addAssistantMessage(message.args.thought));
-    }
-    if (message.args.command) {
-      store.dispatch(
-        addAssistantMessage(
-          `Running this command now: \n\`\`\`\`bash\n${message.args.command}\n\`\`\`\`\nEstimated security risk: ${getRiskText(message.args.security_risk as unknown as ActionSecurityRisk)}`,
-        ),
-      );
-    } else if (message.args.code) {
-      store.dispatch(
-        addAssistantMessage(
-          `Running this code now: \n\`\`\`\`python\n${message.args.code}\n\`\`\`\`\nEstimated security risk: ${getRiskText(message.args.security_risk as unknown as ActionSecurityRisk)}`,
-        ),
-      );
-    } else {
-      store.dispatch(addAssistantMessage(message.message));
-    }
-    return;
+  if ("args" in message && "security_risk" in message.args) {
+    store.dispatch(appendSecurityAnalyzerInput(message));
   }
 
-  if (message.source !== "user" && !message.args?.hidden) {
+  if (message.source !== "user") {
```

Comment:
Maybe we can `message.source === "assistant"` to prevent future bugs (unless there are already other sources?)

Reply:
good call!